server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -163,6 +163,14 @@ public InputStream readBlob(String name) throws IOException {
         }
     }
 
+    @Override
+    public InputStream readBlob(String blobName, long position, int length) throws IOException {
+        final InputStream inputStream = readBlob(blobName);
```

Comment:
I think you could just use this as the default implementation in `BlobContainer`  instead of throwing. I think all our streams support `skip`? :)

Reply:
This is hacky and only less harmful for FS repository so I preferred to implement it in FS blob container and let the other implementation as not supported until the method is correctly implemented for each of them (using range of bytes download)